ci/kind/kind-setup.sh

```diff
@@ -244,25 +245,66 @@ function configure_extra_networks {
   done
 }
 
+# add_kind_ipam_routes add routes for non-ipam test-pods.
+function add_kind_ipam_routes {
+  kubectl get nodes -o jsonpath='{range .items[*]}{.spec.podCIDR}{" "}{.status.addresses[?(@.type=="InternalIP")].address}{"\n"}{end}' |\
+  while read pod_cidr node_ip; do
+    docker_run_with_host_net ip route add "$pod_cidr" via "$node_ip"
+  done
+}
+
+# delete_kind_ipam_routes delete the routes of non-ipam test-pods from the Kind testbed.
+function delete_kind_ipam_routes { 
+  cluster=$(kind get clusters)
```

Comment:
Why doesn't it check if the specified cluster exists but check if any cluster exists?

Reply:
check this comment:https://github.com/antrea-io/antrea/pull/5974#discussion_r1687475828, I need podCIDR to delete the routes from ipam tested and every kind cluster has same podCIDR so if there is any cluster I can use kubectl cmd to get podCIDR and we can check if routes there then delete it.

Reply:
1. for now there will be no multiple kind cluster case, but I am trying to handle one case.
  a.) suppose there is no cluster in the testbed, and according to ipam builder flows are:
      delete kind cluster
      create new one
      run tests
      delete kind cluster
  

to delete kind cluster we run "./ci/kind/kind-setup.sh destroy --all --until 135", lets check this updated function to delete route.

```
function update_kind_ipam_routes {
  local operation="$1"
  kubectl get nodes -o jsonpath='{range .items[*]}{.spec.podCIDR}{" "}{.status.addresses[?(@.type=="InternalIP")].address}{"\n"}{end}' |\
  while read pod_cidr node_ip; do
    docker_run_with_host_net ip route "$operation" "$pod_cidr" via "$node_ip"
  done
}
```
first it will run ```kubectl get nodes....``` cmd to get podCIDR, because we don't have cluster it will show error and exit the script.

```
 ./ci/kind/kind-setup.sh destroy --all --until 135
E0813 06:01:08.798302  973984 memcache.go:265] couldn't get current server API group list: Get "http://localhost:8080/api?timeout=32s": dial tcp 127.0.0.1:8080: connect: connection refused
E0813 06:01:08.799458  973984 memcache.go:265] couldn't get current server API group list: Get "http://localhost:8080/api?timeout=32s": dial tcp 127.0.0.1:8080: connect: connection refused
E0813 06:01:08.800722  973984 memcache.go:265] couldn't get current server API group list: Get "http://localhost:8080/api?timeout=32s": dial tcp 127.0.0.1:8080: connect: connection refused
E0813 06:01:08.801159  973984 memcache.go:265] couldn't get current server API group list: Get "http://localhost:8080/api?timeout=32s": dial tcp 127.0.0.1:8080: connect: connection refused
E0813 06:01:08.802877  973984 memcache.go:265] couldn't get current server API group list: Get "http://localhost:8080/api?timeout=32s": dial tcp 127.0.0.1:8080: connect: connection refused
The connection to the server localhost:8080 was refused - did you specify the right host or port?
```

I think to handle this i can use ```||  true`` after  ```kubectl get nodes``` cmd.

```
ubuntu@a-ms-kind-vms-2004-5-0:~/ipam-enable/antrea$ ./ci/kind/kind-setup.sh destroy --all --until 135
E0813 06:01:08.798302  973984 memcache.go:265] couldn't get current server API group list: Get "http://localhost:8080/api?timeout=32s": dial tcp 127.0.0.1:8080: connect: connection refused
E0813 06:01:08.799458  973984 memcache.go:265] couldn't get current server API group list: Get "http://localhost:8080/api?timeout=32s": dial tcp 127.0.0.1:8080: connect: connection refused
E0813 06:01:08.800722  973984 memcache.go:265] couldn't get current server API group list: Get "http://localhost:8080/api?timeout=32s": dial tcp 127.0.0.1:8080: connect: connection refused
E0813 06:01:08.801159  973984 memcache.go:265] couldn't get current server API group list: Get "http://localhost:8080/api?timeout=32s": dial tcp 127.0.0.1:8080: connect: connection refused
E0813 06:01:08.802877  973984 memcache.go:265] couldn't get current server API group list: Get "http://localhost:8080/api?timeout=32s": dial tcp 127.0.0.1:8080: connect: connection refused
The connection to the server localhost:8080 was refused - did you specify the right host or port?
=== Cleaning up stale kind clusters ===
No kind clusters found.
Deleting external servers
Deleting VLAN subnets
```


2. not sure but what I observed if multiple kind clusters shares same kind network then they have same podCIDR range( but node will different).

```
$ kubectl config get-contexts
CURRENT   NAME          CLUSTER       AUTHINFO      NAMESPACE
          kind-kind     kind-kind     kind-kind     
          kind-kind-1   kind-kind-1   kind-kind-1   
*         kind-kind-2   kind-kind-2   kind-kind-2   
$ kubectl get nodes -o yaml | grep podCIDR
    podCIDR: 10.244.0.0/24
    podCIDRs:
    podCIDR: 10.244.2.0/24
    podCIDRs:
    podCIDR: 10.244.1.0/24
    podCIDRs:
$ kubectl config use-context kind-kind-1
Switched to context "kind-kind-1".
$ kubectl get nodes -o yaml | grep podCIDR
    podCIDR: 10.244.0.0/24
    podCIDRs:
    podCIDR: 10.244.2.0/24
    podCIDRs:
    podCIDR: 10.244.1.0/24
    podCIDRs:
$ kubectl config use-context kind-kind
Switched to context "kind-kind".
$ kubectl get nodes -o yaml | grep podCIDR
    podCIDR: 10.244.0.0/24
    podCIDRs:
    podCIDR: 10.244.2.0/24
    podCIDRs:
    podCIDR: 10.244.1.0/24
    podCIDRs:
$ kubectl get config contexts
error: the server doesn't have a resource type "config"
ubuntu@a-ms-kind-a-0:~/ipam-antrea/antrea$ kubectl config get-contexts
CURRENT   NAME          CLUSTER       AUTHINFO      NAMESPACE
*         kind-kind     kind-kind     kind-kind     
          kind-kind-1   kind-kind-1   kind-kind-1   
          kind-kind-2   kind-kind-2   kind-kind-2   

```